### PR TITLE
Support for `arm-unknown-linux-musleabi(hf)`

### PR DESCRIPTION
**Pull request by justsmth**
### Issues:
Addresses: CryptoAlg-2718

### Description of changes: 
* Adds CI to test builds for the targets `arm-unknown-linux-musleabi` and `arm-unknown-linux-musleabihf`.
* Cleanup cross-rs configurations.
* Allow environment variable to disable slow (long-running) tests, mostly tests that require the generation of large (>2048-bit) RSA keys.

### Callout
* Updated CI configuration to support both FIPS and non-FIPS builds when testing cross-compilation.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: .github/workflows/cross.yml [hunk 1]
```diff
@@ -21,62 +21,67 @@ env:
...
       #   /home/runner/work/aws-lc-rs/aws-lc-rs/aws-lc-sys/aws-lc/crypto/pem/pem_lib.c:707:11: error: 'strncmp' of strings of length 1 and 9 and bound of 9 evaluates to nonzero [-Werror=string-compare]
       #    707 |       if (strncmp(buf, "-----END ", 9) == 0) {
       #        |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
-      - if: ${{ matrix.target == 'riscv64gc-unknown-linux-gnu' }}
+      - if: ${{ matrix.target[0] == 'riscv64gc-unknown-linux-gnu' }}
         run: |
           echo 'AWS_LC_SYS_CFLAGS="-Wno-string-compare"' >> "$GITHUB_ENV"
-      - name: Cross-compilation (build debug)
-        run: cross build -p aws-lc-rs --features unstable --target ${{ matrix.target }}
+      - if: ${{ !startsWith(matrix.target[0], 'x86_64') }}
```
Comment (WillChilds-Klein) [inline .github/workflows/cross.yml:70]: 1. why just `x86_64`?
2. doesn't this mean we lose cross-compilation coverage for `x86_64` targets?

File: .github/workflows/cross.yml [hunk 2]
```diff
@@ -21,62 +21,67 @@ env:
...
+          - [aarch64-unknown-linux-musl, 1]
+          - [arm-linux-androideabi, 0]
+          - [arm-unknown-linux-gnueabihf, 0]
+          - [arm-unknown-linux-musleabi, 1]
+          - [arm-unknown-linux-musleabihf, 1]
+          - [armv7-linux-androideabi, 0]
+          - [armv7-unknown-linux-gnueabihf, 0]
+          - [i686-unknown-linux-gnu, 0]
+          - [powerpc-unknown-linux-gnu, 1]
+          - [powerpc64-unknown-linux-gnu, 1]
```
Comment (WillChilds-Klein) [inline .github/workflows/cross.yml:41]: what's the reasoning for which are tested w/ FIPS and which aren't?

File: docker/linux-cross/Dockerfile [hunk 3]
```diff
@@ -11,4 +11,9 @@ RUN apt-get update && \
     git config --global --add safe.directory '*' && \
     rm -rf /tmp/*
 
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --profile minimal --default-toolchain stable && \
+    . $HOME/.cargo/env && \
+    cargo install --force --locked bindgen-cli && mv $HOME/.cargo/bin/bindgen /usr/bin && \
```
Comment (WillChilds-Klein) [inline docker/linux-cross/Dockerfile:16]: np

```suggestion
    cargo install --force --locked bindgen-cli && \
    mv $HOME/.cargo/bin/bindgen /usr/bin && \
```

File: .github/workflows/cross.yml (hunk 2 above)
Comment (justsmth) [reply .github/workflows/cross.yml:41]: I'm only enabling the FIPS testing for platforms for which customers have requested.

File: .github/workflows/cross.yml (hunk 1 above)
Comment (justsmth) [reply .github/workflows/cross.yml:70]: * This disables the "slow" tests only for platforms that are emulated (i.e., the non-`x86_64` targets). The `x86_64` will continue to run all tests.
* Although we are disabling several slow-running tests, this change should have very minimal effect on overall code coverage. For example, we still test all RSA functionality with 2048-bit keys, and only disable RSA tests that require the generation of keys sized 3072-bit or larger.

**Review by WillChilds-Klein — APPROVED** [review]

**Review by skmcgrail — APPROVED** [review]

**Pull request merged**